### ISSUE TITLE
When can we expect the training code for OWL-ViT?

### ISSUE THREAD

**Issue by JosephKJ** [google-research/scenic#433]
Thanks for your amazing work. Would it be possible to provide an estimate on when the training code would be released?

**Comment by mjlm** [comment]
We're working on releasing the training code but cannot give a precise ETA yet. It will take at least a few more weeks. I'll keep you posted.

**Comment by HITerStudy** [comment]
@mjlm hello, thanks for your fancy work! When the all of codes can be updated? Please offer the full codes which include the configs for experiments published in the paper. Thanks!

**Comment by mjlm** [comment]
We just published the [training code](https://github.com/google-research/scenic/tree/main/scenic/projects/owl_vit#training).
Please let us know if you have any questions.

**Comment by ronhag** [comment]
Hi there! Exciting news about the training code, thanks a lot! 

When could we expect finetuning instructions? Is it mostly similar to training detection from scratch (some smaller learning rate here and there)? Or are there any other tips to be aware of? :) 

thanks so much!

**Comment by mjlm** [comment]
We just added fine-tuning instructions to the [README](https://github.com/google-research/scenic/tree/main/scenic/projects/owl_vit#fine-tuning).

The config is very similar to the from-scratch config. The only crucial difference is that instead of `config.init_from.codebase = 'clip'`, it uses `config.init_from.checkpoint_path = 'path/to/parent/checkpoint'`.

However, some other changes may be necessary to get good fine-tuning result. They depend on the target dataset, so I can make only general recommendations:

* If the dataset is small, very short fine-tuning schedules might be optimal. I would try values for `config.num_training_steps` starting as low as 500 steps and then keep doubling the value until performance is optimal.
* The default learning rate is a good starting point, but it should be tuned. Try going up and down by a factor of 2 or 3.
* Carefully go through the [preprocessing settings](https://github.com/google-research/scenic/blob/main/scenic/projects/owl_vit/configs/clip_b32_finetune.py#L40-L55) and check if they make sense for your application. For example `remove_forbidden_labels` is only needed to measure zero-shot LVIS performance and therefore not included in the fine-tuning config. Also, if you have a dataset with a small, fixed label space, then you should probably get rid of the `add_random_prompts` op and simply add all categories that are not present in an image as negative labels directly in the dataset decoder.
* Input resolution should be tuned. If your dataset contains mostly large objects, a smaller input resolution may lead to better results (and would be faster).

**Comment by pablopalafox** [comment]
Awesome! Thanks @mjlm, super interesting tips. I'll keep them in mind!

**Comment by unrue** [comment]
Hi, 

is there a complete example how to train Owl-vit? From my  dataset having a lot of images, what I have to do? Such commands are quite cryptic. Thanks

**Comment by shivarajabi** [comment]
hi, i have a dataset of only images, no text_queries and i want to use it as image one shot detection at the end, can i finetune the same way ?
